File: Bella/Bella.Services/Database/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Bella.Services.Database
{
    public class Appointment
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public decimal FinalPrice { get; set; }
        
        [Required]
        public DateTime AppointmentDate { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        public bool IsActive { get; set; } = true;

        // Foreign keys
        [Required]
        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        [Required]
        public int HairdresserId { get; set; }
        public virtual User Hairdresser { get; set; } = null!;

        [Required]
        public int StatusId { get; set; }
        public virtual Status Status { get; set; } = null!;

        // Optional service foreign keys
        public int? HairstyleId { get; set; }
        public virtual Hairstyle? Hairstyle { get; set; }

        public int? FacialHairId { get; set; }
        public virtual FacialHair? FacialHair { get; set; }

        public int? DyingId { get; set; }
        public virtual Dying? Dying { get; set; }
    }
}

File: Bella/UI/bella_hairdresser_mobile/lib/model/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  final int id;
  final double finalPrice;
  final DateTime appointmentDate;
  final DateTime createdAt;
  final bool isActive;
  final int userId;
  final String userName;
  final int hairdresserId;
  final String hairdresserName;
  final int statusId;
  final String statusName;
  final int? hairstyleId;
  final String? hairstyleName;
  final double? hairstylePrice;
  @JsonKey(name: 'hairstyleImage')
  final String? hairstyleImage;
  final int? facialHairId;
  final String? facialHairName;
  final double? facialHairPrice;
  @JsonKey(name: 'facialHairImage')
  final String? facialHairImage;
  final int? dyingId;
  final String? dyingName;
  final String? dyingHexCode;

  Appointment({
    this.id = 0,
    this.finalPrice = 0.0,
    required this.appointmentDate,
    required this.createdAt,
    this.isActive = true,
    this.userId = 0,
    this.userName = '',
    this.hairdresserId = 0,
    this.hairdresserName = '',
    this.statusId = 0,
    this.statusName = '',
    this.hairstyleId,
    this.hairstyleName,
    this.hairstylePrice,
    this.hairstyleImage,
    this.facialHairId,
    this.facialHairName,
    this.facialHairPrice,
    this.facialHairImage,
    this.dyingId,
    this.dyingName,
    this.dyingHexCode,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) => _$AppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
